contracts/validator/ForwardValidatorSetAbstract.sol

```diff
@@ -0,0 +1,54 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @title ForwardValidatorSetAbstract abstract contract specifies a forward validator
+ *        set abstract interfaces.
+ */
+contract ForwardValidatorSetAbstract {
```

Comment:
I'm confused; why is this an `Abstract` and not an interface?

Reply:
or how does it compare to `interface ForwardValidatorSetAbstract {` if we want to obtain `SelfProtocore is ForwardValidatorSetAbstract`

I'm also unsure that we only want to expose the "Forward" part of the interface; because now SelfProtocore will have `inValidatorSet` and OriginProtocore won't.

Reply:
Ok. I understood the error of my ways; it s `Protocore is ForwardValidatorSetAbstract`

Thanks; it s resolved for myself